bebraven/platform
# Add Warden authentication strategy for LtiLaunch'ed requests.

Task: https://app.asana.com/0/1174274412967132/1186566037117867

This fixes the issue where you have to enable third party cookies in Chrome incognito windows (and it would fix any other browsers that don't allow third party cookies).

Note the loose end / TODO: https://app.asana.com/0/1174274412967132/1184057808812020

### Test Plan:
- Launch both a Lesson and a Project in Chrome incognito with third party cookies disabled (default setting). They should open and data should be sent to the LRS.
- Do the same with third party cookies enabled and notice how all the logs of `Authenticated user_id = XX using LtiAuthentication::WardenStrategy` go away.
- Try accessing a page with an invalid state query param. It should error.
- `bundle exec rspec`
